VERSIONING.md

```diff
@@ -0,0 +1,119 @@
+# Versioning
+
+This document aims to provide an overview of how VAST is versioned, and what
+significance a version bump carries.
+
+Run the `vast version` command to get VAST's current version, and the versions
+of its major dependencies that it currently uses:
+
+```json
+{
+  "VAST": "v1.0.4-61-geed8cb41a3",
+  "VAST Build Tree Hash": "0eb048d0eeda5be7081f0c3a659ff17f",
+  "CAF": "0.17.6",
+  "Apache Arrow": "6.0.1",
+  "jemalloc": "5.2.1",
+  "plugins": {
+    "compaction": "v1.2.0-61-g9131346958",
+    "matcher": "v2.1.4-23-gfb69e84ade",
+  }
+}
+```
+
+## Version Components
+
+VAST's version number consists of three mandatory and two optional parts:
+`v<major>.<minor>.<patch>[-rc<candidate>][-<tag>]`. Every release on the default
+branch has an associated annotated and gpg-signed tag of the same name.
+
+### Major
+
+A major version increment indicates that the release included breaking changes.
+Such breaking changes include required changes to user configuration, removal of
+previously deprecated features, and updates to the minimum versions of required
+third-party dependencies.
+
+VAST's database format is guaranteed to be forward-compatible between
+consecutive major versions.
+
+### Minor
+
+A minor version increment indicates changes to functionality, e.g., the addition
+of new features and configuration, or the deprecation of existing features and
+configuration.
+
+VAST processes can connect to one another if the major and minor version match.
+
+### Patch
+
+A patch version increment indicates backwards-compatible bug fixes.
```

Comment:
I thought about this, but that would make the version a bit harder to parse. I'm open to changing this.

Reply:
Does anything speak against just releasing `4.52.0`, i.e., have patch explicitly be 0? I think having variable-length versions with options in there are tricky to handle downstream.

From a workflow perspective, we will always decide whether cutting a patch release is important or whether it can wait until the next minor release. In other words, a patch release would be important enough to not wait and thereby indicates an urgent issue.

Reply:
4.52.0 is what's currently in the document. It's simply a matter of preference whether you want to omit the patch version if it is 0.

Reply:
We talked about this in a quick meeting today, and have decided to omit the patch version when talking about releases publicly. I will mention that upfront in the document.